docs/2.7.0/docs/daml-finance/tutorials/instrument-extension-packages/floating-rate-bond.rst

```diff
@@ -0,0 +1,142 @@
+.. Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+Observations (using a floating rate bond)
+#########################################
+
+This tutorial describes how to define observations. It builds on the previous
+:doc:`Time-based Lifecycling <fixed-rate-bond>` tutorial, which uses a fixed rate bond where all
+coupons are pre-defined using a constant annualized rate. In contrast, the coupons of a
+floating rate bond depend on the value of a reference rate for each coupon period. Hence, the
+lifecycling framework requires the future values of the reference rate. This is referred to as
+``Observations``, which is the main topic of this tutorial.
+
+In this tutorial, we are going to:
+
+#. create a floating rate bond instrument and book a holding on it
+#. create an observation of the floating rate, which is used to define the coupon payment
+#. reuse the lifecycle rule and lifecycle event from the fixed rate bond tutorial
+#. process the event to produce the effects of a coupon payment
+#. instruct settlement by presenting a bond holding
+#. settle the resulting batch atomically
+
+This example builds on the previous :doc:`Time-based Lifecycling <fixed-rate-bond>` tutorial script
+in the sense that the same parties, accounts, lifecycle rule and events are used.
+
+Run the Script
+**************
+
+The code for this tutorial can be executed via the ``runFloatingRateBond`` function in the
+``FloatingRateBond.daml`` module.
+
+Instrument and Holding
+======================
+
+For the purpose of showcasing the
+:ref:`Observation <module-daml-finance-data-numeric-observation-78761>` concept,
+we need a suitable sample instrument.
+:ref:`Floating rate bonds <module-daml-finance-instrument-bond-floatingrate-instrument-98586>`
+pay a coupon which is determined by a reference rate, e.g. 3M Libor. The
+:doc:`Bond Extension tutorial <bond-extension>` describes this instrument in more detail. Here, we
+briefly show how to create the bond instrument using a factory:
+
+.. literalinclude:: ../../finance-lifecycling/daml/Scripts/FloatingRateBond.daml
+  :language: daml
+  :start-after: -- CREATE_FLOATING_RATE_BOND_INSTRUMENT_BEGIN
+  :end-before: -- CREATE_FLOATING_RATE_BOND_INSTRUMENT_END
+
+Compared to the fixed rate bond, notice that this floating rate instrument also has a
+``referenceRateId``, that specifies which ``Observations`` to use in the lifecycling section below.
+
+We also create a bond holding in Bob's account:
+
+.. literalinclude:: ../../finance-lifecycling/daml/Scripts/FloatingRateBond.daml
+  :language: daml
+  :start-after: -- CREATE_FLOATING_RATE_BOND_HOLDING_BEGIN
+  :end-before: -- CREATE_FLOATING_RATE_BOND_HOLDING_END
+
+A holding represents the ownership of a certain amount of an :ref:`instrument <instrument>` by an
```

Comment:
Not sure if it's worth repeating it here, as we expect this to be clarified after the getting started section